app/pods/search-schema/edit/route.js

```diff
@@ -0,0 +1,55 @@
+import Ember from 'ember';
+
+export default Ember.Route.extend({
+    model(params) {
+        return this.explorer.getCluster(params.clusterId, this.store)
+            .then(function(cluster){
+                return cluster.get('searchSchemas').findBy('name', params.searchSchemaId);
+            });
+    },
+
+    afterModel(model, transition) {
+        if (!model.get('content')) {
+            return Ember.$.ajax({
+                type: 'GET',
+                url: model.get('url'),
+                dataType: 'xml'
+            }).then(function(data) {
+                let xmlString = (new XMLSerializer()).serializeToString(data);
+                model.set('content', xmlString);
+            });
+        }
+    },
+
+    actions: {
+        updateSchema: function(schema) {
+            let xmlString = schema.get('content');
+            let self = this;
+            let xmlDoc = null;
+            let clusterId = schema.get('cluster').get('id');
+            let schemaId = schema.get('name');
+
+            try {
+                xmlDoc = Ember.$.parseXML(xmlString);
+            } catch(error) {
+                // TODO: Put in proper error messaging
+                alert('Invalid XML. Please check and make sure schema is valid xml.');
```

Comment:
> Something like a modal instead of using alert

Yeah, I've been thinking about that as well, that we need some either modal windows, or (better yet) some message/error divs under the crumb trail, etc.

Alert is totally fine for now though.